kingpin/actors/aws/base.py

```diff
@@ -83,19 +85,34 @@ def __init__(self, *args, **kwargs):
 
         super(AWSBaseActor, self).__init__(*args, **kwargs)
 
-        if not (aws_settings.AWS_ACCESS_KEY_ID and
+        # By default, we will try to let Boto handle discovering its
+        # credentials at instantiation time. This _can_ result in synchronous
+        # API calls to the Metadata service, but those should be fast.
+        key = None
+        secret = None
+
+        # In the event though that someone has explicitly set the AWS access
+        # keys in the environment (either for the purposes of a unit test, or
+        # because they wanted to), we use those values.
+        if (aws_settings.AWS_ACCESS_KEY_ID and
```

Comment:
Do you have any thoughts yet on this? I think you were thinking about how to mock out the environment during the tests?